lib/ansible/module_utils/common/dict_transformations.py

```diff
@@ -118,7 +118,10 @@ def dict_merge(a, b):
         if k in result and isinstance(result[k], dict):
             result[k] = dict_merge(result[k], v)
         else:
-            result[k] = deepcopy(v)
+            if isinstance(result[k], list):
```

Comment:
This fails when `k` is not in `result`. 

Probably want:

```
if k in result:
    if isinstance(result[k], dict):
        result[k] = dict_merge(result[k], v)
    elif isinstance(result[k], list):
        result[k] = [dict_merge(result[k][i], v[i]) if isinstance(result[k][i], dict) else deepcopy(v[i]) for i in range(len(result[k]))]
    else:
        result[k] = deepcopy(v)
else:
    result[k] = deepcopy(v)
```

Reply:
Or perhaps:
```
if isinstance(result.get(k), dict):
    result[k] = dict_merge(result[k], v)
elif isinstance(result.get(k), list):
    result[k] = [dict_merge(result[k][i], v[i]) if isinstance(result[k][i], dict) else deepcopy(v[i]) for i in range(len(result[k]))]
else:
    result[k] = deepcopy(v)
```

Reply:
It's not that simple actually. I've been testing the patch in my roles/playbooks and quickly encountered "index out of bound" issue when changing "ports" list members order in my k8s definitions. I had to change "for i in range(len(result[k]))]" to for i in range(min(len(result[k]), len(v))], and it resulted in diff missing some changes when doing more changes to k8s definition lists (I was experimenting with list of pod envvars: removed some and added some more).